listenbrainz/webserver/static/js/src/utils.tsx

```diff
@@ -240,17 +240,25 @@ const preciseTimestamp = (
 
   // determine which display setting based on time difference to use if no argument was provided
   if (!display) {
-    const msDifference = new Date().getTime() - listenDate.getTime();
-    // over one year old : show with year
-    if (msDifference / (1000 * 3600 * 24 * 365) > 1) {
-      display = "includeYear";
+    const now = new Date();
+    const currentYear = now.getUTCFullYear();
+    const listenYear = listenDate.getUTCFullYear();
+    // Date is today : format using timeago
+    if (
+      now.getUTCDate() === listenDate.getUTCDate() &&
+      now.getUTCMonth() === listenDate.getUTCMonth() &&
+      currentYear === listenYear
+    ) {
+      display = "timeAgo";
```

Comment:
Because you're comparing with UTC dates here, the switchover from timeAgo to excludeYear may happen at midday for someone in NZ (right?). I think it might be better if we converted in the browser's timezone...